orchestra/utils/view_helpers.py

```diff
@@ -12,3 +14,93 @@ class IsAssociatedWorker(permissions.BasePermission):
     def has_object_permission(self, request, view, obj):
         worker = Worker.objects.get(user=request.user)
         return obj.worker == worker
+
+
+def _get_changed_fields(old_todo, new_todo):
+    changed_fields = []
+    if old_todo.title != new_todo.title:
+        changed_fields.append('title')
+    if old_todo.details != new_todo.details:
+        changed_fields.append('details')
+    dict1 = {}
+    dict2 = {}
+    if isinstance(old_todo.additional_data, dict):
+        dict1 = old_todo.additional_data
+    if isinstance(new_todo.additional_data, dict):
+        dict2 = new_todo.additional_data
+    changed_subfields = [k for k in dict1.keys()
+                         if dict1.get(k) != dict2.get(k)]
+    changed_fields.extend(changed_subfields)
+    return changed_fields
+
+
+def get_todo_change(old_todo, new_todo):
+    # When activity_log is updated, `todo_change = None`
+    # to avoid triggering any slack messages
+    todo_change = ''
+    changed_fields = _get_changed_fields(old_todo, new_todo)
+    if old_todo.completed != new_todo.completed:
+        todo_change = 'complete' if new_todo.completed else 'incomplete'
+    elif old_todo.skipped_datetime != new_todo.skipped_datetime:
+        todo_change = 'not relevant' \
+            if new_todo.skipped_datetime else 'relevant'
+    message = ''
+    if len(todo_change) and len(changed_fields):
```

Comment:
after some light updates of the text in the comment below, do some light QA of the if/elif/else here to make sure the messages are understandable.